backend/src/main/java/de/bund/digitalservice/ris/norms/application/service/ReleaseService.java

```diff
@@ -19,22 +31,149 @@ public class ReleaseService implements ReleaseAnnouncementUseCase {
+      }
+
+      analysis
+        .get()
+        .getActiveModifications()
+        .stream()
+        .map(TextualMod::getDestinationHref)
+        .flatMap(Optional::stream)
+        .map(Href::getExpressionEli)
+        .flatMap(Optional::stream)
+        .map(eli -> normService.loadNorm(new LoadNormUseCase.Query(eli)))
+        .forEach(normsToPublish::add);
+    }
+
+    // Delete the files from a previous release of the same announcement if they are still queued for publishing.
+    // This is to prevent residual files from remaining if some time boundaries changed and this release will create
+    // different expressions.
+    normsToPublish.forEach(norm ->
+      deleteQueuedNormsPort.deleteQueuedForPublishNorms(
+        new DeleteQueuedNormsPort.Command(norm.getWorkEli())
+      )
+    );
+
+    // generate all future versions for all norms that will be published
+    Set<Norm> allVersionsOfAllNormsToPublish = new HashSet<>();
+    for (Norm norm : normsToPublish) {
+      // We need to publish the manifestation of the norm with the correct "nachfolgende-version-id", therefore we
+      // use the manifestation created from createNewExpression of the next time boundary if it exists
+      Norm latestNormExpression = createNewVersionOfNormService.createNewManifestation(norm);
+
+      var dates = norm
+        .getMeta()
+        .getAnalysis()
+        .stream()
+        .flatMap(analysis -> analysis.getPassiveModifications().stream())
+        .flatMap(textualMod -> textualMod.getForcePeriodEid().stream())
+        .map(norm::getStartDateForTemporalGroup)
+        .flatMap(Optional::stream)
+        .map(LocalDate::parse)
+        .distinct()
+        .sorted();
+
+      for (LocalDate date : dates.toList()) {
+        var result = createNewVersionOfNormService.createNewExpression(latestNormExpression, date);
+        allVersionsOfAllNormsToPublish.add(result.newManifestationOfOldExpression());
+
+        latestNormExpression =
+        timeMachineService.applyPassiveModifications(
+          new ApplyPassiveModificationsUseCase.Query(
+            result.newExpression(),
+            date.atStartOfDay(ZoneId.systemDefault()).toInstant()
+          )
+        );
+      }
+
+      allVersionsOfAllNormsToPublish.add(latestNormExpression);
+    }
+
+    allVersionsOfAllNormsToPublish.forEach(norm ->
+      norm.setPublishState(NormPublishState.QUEUED_FOR_PUBLISH)
+    );
+    allVersionsOfAllNormsToPublish.forEach(norm ->
+      updateOrSaveNormPort.updateOrSave(new UpdateOrSaveNormPort.Command(norm))
+    );
+
+    // Create new manifestations for the expressions that are directly targeted by this announcement (so not the once
+    // only created by the time-machine, but all the other once) so the newest manifestation is still unpublished and
+    // can be edited further
+    normsToPublish.forEach(norm -> {
+      // We need the new manifestations to use tomorrow's date so they do have a different eli to the once just
```

Comment:
Is this really needed? If on the same day, after publishing, more changes are done to the same annoucement, won't be working on the same manifestations we had created already (amending law and ZF0) and then just override the QUEUED manifestations of the future versions?

Reply:
our application assumes that the zf0 is the newest manifestation of the mentioned expression. But the publishing process created new manifestations for those expressions with dates that are newer than the date of the manifestation of the zf0 and amending norm previously used for editing. So we would try to overwrite the queued files. Therefore we need new manifestations with a newer date so the newest manifestations are still unpublished.